model/qti/identifierGenerator/SimpleQtiIdentifierGenerator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2024 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\qti\identifierGenerator;
+
+use InvalidArgumentException;
+
+class SimpleQtiIdentifierGenerator implements IdentifierGenerator
+{
+    public function generate(array $options = []): string
+    {
+        $resourceId = $options[self::OPTION_RESOURCE_ID] ?? null;
+
+        if (!is_string($resourceId)) {
+            throw new InvalidArgumentException(
+                'QTI Identifier generation failure: resource ID is required and must be a string'
+            );
+        }
+
+        $prefix = LOCAL_NAMESPACE . '#';
+        $resourceId = $options['resourceId'];
+
+        // If we have an item from the current environment
```

Comment:
- [ ] @shpran It is generally recommend to move to a private explanatory method, rather than comments
- [ ] If I am not mistaken we have already services that generate the URI id, deal with namespaces, etc. Are we sure we need this logic here? Can't we use existing services for that?

```

```

Reply:
This logic was in private method and was used only for QTI creation.
Also, this will be removed, because we will not sync property with QTI ID, so I can use unique ID generation separately without other generators (just to generate property if FF enabled)

Reply:
For URI logic we already have the `generis/uriProvider` and the `core_kernel_uri_UriService`, lets not duplicate sensitive logic here, please